src/system.rs

```diff
@@ -31,3 +127,88 @@ mod unit_tests {
         assert_eq!(uptime_internal("12489513.08 22906637.29\n"), 12489513);
     }
 }
+
+/// Returns information about virtual memory usage
+///
+/// `/proc/meminfo` contains the virtual memory statistics
+pub fn virtual_memory() -> Option<VirtualMemory> {
```

Comment:
I think it would be better for this to return https://doc.rust-lang.org/std/result/ - that means the error can give more information to the caller about what went wrong:

- /proc/meminfo not present
- MemAvailable not present
- Shmem not present